components/browser/state/src/main/java/mozilla/components/browser/state/reducer/TabListReducer.kt

```diff
@@ -196,36 +199,41 @@ internal object TabListReducer {
             is TabListAction.RemoveAllTabsAction -> {
                 state.copy(
                     tabs = emptyList(),
-                    selectedTabId = null
+                    selectedTabId = null,
+                    tabPartitions = emptyMap()
                 )
             }
 
             is TabListAction.RemoveAllPrivateTabsAction -> {
                 val selectionAffected = state.selectedTab?.content?.private == true
-                val updatedTabs = state.tabs.filterNot { it.content.private }
+                val partition = state.tabs.partition { it.content.private }
```

Comment:
We need to separate the tabs we remove from the ones we keep. So, using the built-in `partition` call :)